src/main/groovy/com/twcable/grabbit/client/batch/steps/jcrnodes/JcrNodesWriter.groovy

```diff
@@ -88,8 +88,9 @@ class JcrNodesWriter implements ItemWriter<ProtoNode>, ItemWriteListener {
     }
 
     private static void writeToJcr(ProtoNode nodeProto, Session session) {
-        JcrNodeDecorator jcrNode = new ProtoNodeDecorator(nodeProto).writeToJcr(session)
-        jcrNode.setLastModified()
+        ProtoNodeDecorator protoNodeDecorator = new ProtoNodeDecorator(nodeProto)
+        JcrNodeDecorator nodeDecorator = new JcrNodeDecorator(protoNodeDecorator.getOrCreateNode(session))
+        nodeDecorator.writeToJcr(protoNodeDecorator, session)
         // This will processed all mandatory child nodes only
         if(nodeProto.mandatoryChildNodeList && nodeProto.mandatoryChildNodeList.size() > 0) {
```

Comment:
From what I thought was that all the JCR operations should be under the Jcr decorator classes and Proto Decorators should be used for proto related operations only.. In that case, moving this to ProtoNodeDecorator would mean creating a JcrNodeDecorator there and passing necessary values to jcr decorator methods..